Morpho borrow strategies

paszkowskiDamian opened the pull request:
## Ticket URL

Please provide a link to the ticket:

## Description of Changes

Please list the changes introduced by this PR:

- [ ] add view for morpho position 
- [ ] add borrow strategies for morpho, open/deposit-borrow/withdraw-repay 
- [ ] add cli commands to test strategies

## How to Test

Please provide instructions on how to test the changes in this PR:

## PR Definition of Done

Please ensure the following requirements have been met before marking the PR as ready for review:

- [ ] All checks are passing
- [ ] PR is linked to a corresponding ticket
- [ ] PR title is clear and concise
- [ ] Code has been self-reviewed and any fixes or improvements noted (See Code review standards in Notion)
- [ ] Documentation has been updated if necessary

paszkowskiDamian committed:
feat: add abis

paszkowskiDamian committed:
feat: add morpho position

paszkowskiDamian committed:
chore: add test command

paszkowskiDamian committed:
feat: finish strategies

paszkowskiDamian committed:
fix: payback and withdraw

paszkowskiDamian committed:
fix: debt amount

paszkowskiDamian committed:
feat: add validation

paszkowskiDamian committed:
Merge branch 'dev' of github.com:OasisDEX/oasis-earn-sc into damianpaszkowski/sc-12957/morpho-views

jakubswierczek commented on packages/dma-library/src/strategies/morphoblue/borrow/deposit-borrow.ts:
```diff
@@ -0,0 +1,121 @@
+      notices: [],
+      successes: [],
+      targetPosition,
+      position: targetPosition,
```
`position` and `targetPosition` should be the same?

jakubswierczek commented on packages/dma-library/src/strategies/morphoblue/borrow/payback-withdraw.ts:
```diff
@@ -0,0 +1,115 @@
+export type MorphoPaybackWithdrawStrategy = (
+  args: MorphobluePaybackWithdrawPayload,
+  dependencies: MorphoBlueCommonDependencies,
+) => Promise<AjnaStrategy<MorphoBluePosition>>
```
`AjnaStrategy` ?

paszkowskiDamian committed:
v0.5.29-morpho

paszkowskiDamian committed:
refactor

paszkowskiDamian committed:
Merge branch 'dev' of github.com:OasisDEX/oasis-earn-sc into damianpaszkowski/sc-12957/morpho-views

paszkowskiDamian committed:
fix: some building issues

paszkowskiDamian committed:
lint

paszkowskiDamian committed:
fix

paszkowskiDamian committed:
v0.5.33-morpho

paszkowskiDamian committed:
fix

jakubswierczek committed:
Update package manager and Node.js version

The package manager has been specified as "yarn@1.22.1" in each package.json file, and the Node.js version has been updated to 18.12 in .nvmrc. Additionally, some changes have been made to the structure and dependencies in deploy-configurations/package.json and domain/package.json.

jakubswierczek committed:
Upgrade ethers dependency and refactor imports

This update upgrades the "ethers" dependency to the latest version (from 5.6.2 to 5.7.2) across all the relevant package.json files for better performance and reliability. Also, some unnecessary import sequences in multiple files were refactored for better code readability and organization.

jakubswierczek committed:
Merge branch 'dev' into damianpaszkowski/sc-12957/morpho-views

# Conflicts:
#	packages/dma-library/package.json

jakubswierczek committed:
v0.5.35-rc.1

jakubswierczek committed:
Update "@oasisdex/dma-library" version and remove unnecessary dependency

In this commit, the version of "@oasisdex/dma-library" is updated from "^0.5.18" to "^0.5.35-rc.1" in summer-cli's package.json file. The redundant dependency, "@oasisdex/dma-library" version "0.5.34", is also removed from the yarn.lock file improving the overall project dependency tree.

paszkowskiDamian committed:
fix

jakubswierczek committed:
Add prebuild and prelint scripts to package.json

This commit introduces "prebuild" and "prelint" scripts to the "scripts" section of the package.json file to ensure typechain is run before these tasks. Additionally, "lint" script is modified to include the running of the "prelint" script.

paszkowskiDamian committed:
fix: remove dubble transfer

paszkowskiDamian committed:
Merge branch 'damianpaszkowski/sc-12957/morpho-views' of github.com:OasisDEX/oasis-earn-sc into damianpaszkowski/sc-12957/morpho-views

robercano committed:
fix: add fake WETH to service registry for tests

jakubswierczek committed:
Merge branch 'dev' into damianpaszkowski/sc-12957/morpho-views

paszkowskiDamian committed:
tests

paszkowskiDamian committed:
Merge branch 'damianpaszkowski/sc-12957/morpho-views' of github.com:OasisDEX/oasis-earn-sc into damianpaszkowski/sc-12957/morpho-views

paszkowskiDamian committed:
lint
